test/test_mahalanobis_mixin.py

```diff
@@ -150,12 +150,13 @@ def check_is_distance_matrix(pairwise):
   assert np.array_equal(pairwise, pairwise.T)  # symmetry
   assert (pairwise.diagonal() == 0).all()  # identity
   # triangular inequality
-  for i in range(pairwise.shape[1]):
-    # since we already checked symmetry we can start at i
-    for j in range(i, pairwise.shape[1]):
-      for k in range(pairwise.shape[1]):
-        assert (pairwise[i, j] - (pairwise[i, k] + pairwise[k, j]) <= 0 +
-                1e-3).all()
+  rows, cols = pairwise.shape
+  tol = np.finfo(float).eps
```

Comment:
Maybe here we can start with the smaller value possible, and if some distance function fails to pass the test, we could investigate more and decide what is a reasonable value ?